### ISSUE TITLE
Infinite loop reported when trying to layout a multiline array

### ISSUE THREAD

**Issue by terrywbrady** [rubocop/rubocop#12548]
I would like rubocop to reformat arrays in long lines of code with specific indention.

##.rubocop.yml
```yml
Layout/FirstArrayElementIndentation:
  Enabled: true
Layout/FirstArrayElementLineBreak:
  Enabled: true
Layout/ArrayAlignment:
  EnforcedStyle: with_fixed_indentation
Layout/MultilineArrayBraceLayout:
  EnforcedStyle: new_line

AllCops:
  NewCops: enable
```

## robo.rb
```rb
# frozen_string_literal: true

# test an issue with robocop rules
class RoboTest
  def initialize
    puts(['aaaaaaaaaaa', 'bbbb', 'the quick brown fox jumped over the lazy dogs', 'one two three four five six seven eight nine ten' ])
  end
end
```
## No error occurs without auto correct

```
bash-4.2$ rubocop robo.rb
Inspecting 1 file
C

Offenses:

robo.rb:6:121: C: [Correctable] Layout/LineLength: Line is too long. [135/120]
    puts(['aaaaaaaaaaa', 'bbbb', 'the quick brown fox jumped over the lazy dogs', 'one two three four five six seven eight nine ten' ])
                                                                                                                        ^^^^^^^^^^^^^^^
robo.rb:6:133: C: [Correctable] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
    puts(['aaaaaaaaaaa', 'bbbb', 'the quick brown fox jumped over the lazy dogs', 'one two three four five six seven eight nine ten' ])
                                                                                                                                    ^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

## Error occurs with auto-correct

The process returns an infinite loop error if I try to autocorrect.

I have seen this error with 1.31.2 and 1.59.0.

Please see https://stackoverflow.com/questions/77669036/rubocop-indention-layout-of-multiline-array-failing-with-infinite-loop-error

## Expected behavior

I expected the auto-correct to either apply the rules I provided or leave the code unchanged.

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

```
bash-4.2$ rubocop --debug -a robo.rb
For /home/tbrady/mrt-cron/coll-health-obj-analysis: configuration from /home/tbrady/mrt-cron/coll-health-obj-analysis/.rubocop.yml
Default configuration from /home/tbrady/.gem/ruby/gems/rubocop-1.59.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/tbrady/mrt-cron/coll-health-obj-analysis/robo.rb
C

Offenses:

robo.rb:7:7: C: [Corrected] Layout/FirstArrayElementIndentation: Use 2 spaces for indentation in an array, relative to the first position after the preceding left parenthesis.
      'aaaaaaaaaaa', 'bbbb', 'the quick brown fox jumped over the lazy dogs',
      ^^^^^^^^^^^^^
robo.rb:7:12: C: [Corrected] Layout/ArrayAlignment: Use one level of indentation for elements following the first line of a multi-line array.
           'aaaaaaaaaaa', 'bbbb', 'the quick brown fox jumped over the lazy dogs',
           ^^^^^^^^^^^^^

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in /home/tbrady/mrt-cron/coll-health-obj-analysis/robo.rb and caused by Layout/FirstArrayElementIndentation -> Layout/ArrayAlignment
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:332:in `check_for_infinite_loop'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:315:in `block in iterate_until_no_changes'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:314:in `loop'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:163:in `file_offenses'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:154:in `process_file'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `each'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `reduce'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:120:in `inspect_files'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/runner.rb:73:in `run'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli/command.rb:11:in `run'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli/environment.rb:18:in `run'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli.rb:118:in `run_command'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli.rb:125:in `execute_runners'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli.rb:51:in `block in run'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/lib/rubocop/cli.rb:43:in `run'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/exe/rubocop:19:in `block in <top (required)>'
/usr/share/ruby/benchmark.rb:308:in `realtime'
/home/tbrady/.gem/ruby/gems/rubocop-1.59.0/exe/rubocop:19:in `<top (required)>'
/home/tbrady/bin/rubocop:25:in `load'
/home/tbrady/bin/rubocop:25:in `<main>'
Finished in 0.5349028579657897 seconds
```

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

Fortunately, this error seems easy to reproduce with a very small source file.
- Create the `.rubocop.yml` file described above
- Create the `robo.rb` file described above
- Run `rubocop -a robo.rb`

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
bash-4.2$ rubocop -V
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.0.6) [x86_64-linux]
```
